fix: Git + Datasource Environments (#1556)
appsmithorg/appsmith-docs: website/docs/connect-data/how-to-guides/setup-datasource-environments.md

Developers can switch between staging and production environments in the *Edit mode*. In addition, users with the necessary permissions can also switch to the staging environment configuration when accessing the app in *View mode*. For more information on permissions, you can refer to the [Granular Access Control.](/advanced-concepts/granular-access-control).

With this, you can create queries and switch between environments. As you toggle between environments, the queries would automatically use the selected environment.

## Use Git with Datasource Environments

Using Git with environment configurations allows you to connect your application to version control and work on different branches, ensuring that changes made during development do not affect the production environment. This approach enables you to thoroughly test changes on staging data before merging them into the master branch for deployment.

You can create a new branch for each feature you're working on and commit changes to that branch as you make progress. Once a feature is stable and ready for deployment, you can merge it into the master branch.

To use environments with Git, you can follow these steps:

> 1. If you haven't done it already, [connect your application to Git](/advanced-concepts/version-control-with-git).

Reviewer: ```suggestion
1. If you haven't done it already, [connect your application to Git](/advanced-concepts/version-control-with-git/connecting-to-git-repository).
```

> 2. Connect to a datasource and configure its staging and production environments.

Reviewer: ```suggestion
2. Connect to a datasource and configure the _Staging_ and _Production_ environments.
```

> 3. On the staging environment, create and switch to a `feature` [branch](/advanced-concepts/version-control-with-git/working-with-branches) to work on specific enhancements or features. 

Reviewer: ```suggestion
3. In the _Staging_ environment, create and switch to a `feature` branch. See [Setup Branches](/advanced-concepts/version-control-with-git/working-with-branches) to create a branch.
```

> 4. Use the staging environment while working on your application so that you can test the changes without affecting production data.

Reviewer: ```suggestion
4. Use the _Staging_ environment to make edits to your application so that you can test the changes without affecting production data.
```
